### ISSUE TITLE
OneNote offline file format support request

### ISSUE THREAD

**Issue by Lokawn** [nico9889/OneNote2XournalppAddon#3]
thanks for the add-on.
I was thinking is there any plan for adding ability to convert Onenote files downloaded using onedrive, I saw your earlier project  which worked on offline files, albeit html, and this came to my mind.
Cheers

**Comment by nico9889** [comment]
I'm glad that the addon works and it's useful to you, thanks for you feedback!
Currently I have no plan to support OneNote file conversion in the near future.
There are a few reasons:

#### I want to keep the process as simple as possible
I can't find any "easy" official way to download OneNote files, I tried from OneDrive app on Windows, Android and Web but there's no way to download these files. I didn't try unofficial clients like onedrive-cli, if I'm missing something please let me know.
As you said, I made a script in the past that permitted to convert HTML files that kind of worked, but had two major issues that made me decide to make this addon:
* static HTML is missing some offset values that were calculated only by the browser while rendering the page;
* **the process to "obtain" the HTML wasn't user friendly at all and it was very annoying, even for me that I made the script**.

#### I see only a few advantages on doing that
The only advantage that I can see is that the OneNote file, standing on the [documentation](https://docs.microsoft.com/en-us/openspecs/office_file_formats/ms-one/73d22548-a613-4350-8c23-07d15576be50), should contain the entire NoteBook so all the notes would be converted with just one click, but this adds another level of complexity because it would require a proper user interface to select which page to convert.

**TL;DR:** I don't think that the advantages would justify the effort on supporting the file format right now.

I may try to do something if, for example, this addon stops working permanently due to a Microsoft OneNote or Web Browser update and there's no way to fix it, depending on the file format availability in the future since it looks like Microsoft is removing any way to distribute it.

I'll keep this issue open as a reminder or just in case someone passing, by reading this finds, the idea more appealing than me.

By the way, I changed the title of your issue to make the issue/request evident.

**Comment by Lokawn** [comment]
> I can't find any "easy" official way to download OneNote files, I tried from OneDrive app on Windows, Android and Web but there's no way to download these files.

Oh I just downloaded the zip of whole Documents folder from onedrive web, which contains the OneNote notebooks.
It contains different files which contain just raw data, I don't think they can be of any use but attaching them here in case you change your mind or someone else finds this.
[This is the zip which I downloaded.](https://github.com/nico9889/OneNote2XournalppAddon/files/9499015/Documents-Onedrive.zip)
This is how the it looks (I converted xopp file to png as I can't upload xopp file here).
![OneNote](https://user-images.githubusercontent.com/39772448/188693257-f8b62d70-4e70-4efd-961d-f18d877aadc8.png)

**Comment by nico9889** [comment]
This issue has been open for quite a while now, and I think it's time to close it.

As already noted, this request falls outside the scope of this add-on for two main reasons:
* it would require a standalone application, not a simple add-on,as the browser environment is too restrictive for this task.
* I want to maintain a streamlined experience. While the results may not be "perfect," the add-on currently only requires a few clicks to produce a usable file.

If you are still interested in applications that have support for .one files, RNote developers are working on it and slowly making some progress (https://github.com/flxzt/rnote/issues/744).

Have a nice day 👋
